### PR TITLE
make port discovery automatic & update java ver for 1.17

### PR DESCRIPTION
**Pull request by u9g**

### CODE REVIEW

File: test/clientTest.js [hunk 1]
```diff
@@ -28,28 +39,45 @@ for (const supportedVersion of mc.supportedVersions) {
   describe('client ' + version.minecraftVersion, function () {
     this.timeout(10 * 60 * 1000)
 
-    before(download.bind(null, version.minecraftVersion, MC_SERVER_JAR))
+    before(async function () {
+      this.timeout(30 * 1000)
+      await new Promise((resolve, reject) => {
+        wrap.deleteServerData(err => {
```
Comment (rom1504) [inline test/clientTest.js:45]: Why are you deleting server data in before ?

File: test/clientTest.js [hunk 2]
```diff
@@ -172,13 +200,14 @@ for (const supportedVersion of mc.supportedVersions) {
         })
       })
     })
-
-    describe('online', function () {
+    /*
```
Comment (rom1504) [inline test/clientTest.js:203]: Why comment instead of skip ? It disables standard

**Comment by rom1504** [comment]
Thanks, put 2 small comments

File: test/clientTest.js (hunk 2 above)
Comment (u9g) [reply test/clientTest.js]: fixed in 8daf006

File: test/serverTest.js [hunk 3]
```diff
@@ -379,9 +379,9 @@ for (const supportedVersion of mc.supportedVersions) {
           entityId: client.id,
           levelType: 'default',
           gameMode: 1,
-          previousGameMode: 255,
+          previousGameMode: version.version >= 755 ? 0 : 255,
           worldNames: ['minecraft:overworld'],
-          dimensionCodec: (version.version >= 735 ? mcData.loginPacket.dimension : { name: '', type: 'compound', value: { dimension: { type: 'list', value: { type: 'compound', value: [w] } } } }),
+          dimensionCodec: version.version >= 755 ? mcData.loginPacket.dimensionCodec : (version.version >= 735 ? mcData.loginPacket.dimension : { name: '', type: 'compound', value: { dimension: { type: 'list', value: { type: 'compound', value: [w] } } } }),          dimension: (version.version >= 735 ? mcData.loginPacket.dimension : 0),
```
Comment (rom1504) [inline test/serverTest.js:384]: you got dimension twice there, look at the end of the line

File: test/serverTest.js [hunk 4]
```diff
@@ -328,9 +328,9 @@ for (const supportedVersion of mc.supportedVersions) {
...
```
Comment (rom1504) [inline test/serverTest.js:333]: there too

File: test/serverTest.js (hunk 3 above)
Comment (u9g) [reply test/serverTest.js:384]: OHHHHHH

**Comment by rom1504** [comment]
one last comment

**Review by rom1504 — APPROVED** [review]

**Pull request merged**